packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -99,9 +124,108 @@ const CompactInteractiveList = (props) => {
+      // currently a cell can have only one interact-able element in it, which gets auto-focused.
+      interactiveChildren[0].focus();
+    } else {
+      // cell gets focus if there is no interact-able elements in it.
+      focusedCellElement.focus();
+    }
+  };
+
+  const setFocusedRowCol = ({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    setFocusOnCell({ row, cell });
+  };
+
+  const handleOnCellSelect = useCallback(({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    // check for interactive elements in that cell
+    // add 1 to the row number to accomodate for hidden header
+    const elementToFocus = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(elementToFocus);
+    // if no interactive elements
+    if (onCellSelect && interactiveChildren?.length === 0) {
+      onCellSelect(rows[row].id, columns[cell].id);
+    }
+  }, [onCellSelect, rows, columns]);
+
+  const handleKeyDown = (event) => {
+    let moveFocusTo = focusedCell.current;
+
+    const targetElement = event.target;
+
+    // Allow default behavior if the event target is an editable field
+
+    if (event.keyCode !== KeyCode.KEY_TAB
+        && (isTextInput(targetElement)
+            || ['textarea', 'select'].indexOf(targetElement.tagName.toLowerCase()) >= 0
+            || (targetElement.hasAttribute('contentEditable') && targetElement.getAttribute('contentEditable') !== false))) {
+      return;
+    }
+
+    const key = event.keyCode;
+    switch (key) {
+      case KeyCode.KEY_UP:
+        moveFocusTo = handleUpKey(focusedCell.current, numberOfColumns, flowHorizontally, rows.length);
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_DOWN:
+        moveFocusTo = handleDownKey(focusedCell.current, numberOfColumns, flowHorizontally, rows.length);
+        event.preventDefault();
+        break;
+      case KeyCode.KEY_LEFT: {
+        moveFocusTo = handleLeftKey(event, focusedCell.current, numberOfColumns, flowHorizontally, columns.length, rows.length);
+        break;
+      }
+      case KeyCode.KEY_RIGHT: {
+        moveFocusTo = handleRightKey(event, focusedCell.current, numberOfColumns, flowHorizontally, columns.length, rows.length);
+        break;
+      }
+      case KeyCode.KEY_SPACE:
+        if (onCellSelect) {
+          handleOnCellSelect(focusedCell.current);
+        }
+        event.preventDefault();
+        return;
+      case KeyCode.KEY_ESCAPE:
+        if (onClearSelection) {
+          onClearSelection();
+        }
+        event.preventDefault();
+        return;
+      case KeyCode.KEY_TAB:
+        moveFocusFromElement(listRef, id, !event.shiftKey);
+        event.preventDefault();
+        return;
+      default:
+        return;
+    }
+
+    setFocusedRowCol(moveFocusTo);
+    event.preventDefault(); // prevent the page from moving with the arrow keys.
+  };
+
+  const onFocus = (event) => {
```

Comment:
Do we need the focus event listener for all cases here?  This event is triggered event when focus is moving inside the control because the focus event bubbles in React.

Reply:
Great catch, fixed focus to not be triggered by children swap in https://github.com/cerner/terra-framework/pull/1919/commits/092e8379aaec6fda0ec2087e958f6f7dcdd8ae37